src/v/iceberg/time_transform_visitor.h

```diff
@@ -23,4 +23,23 @@ struct hour_transform_visitor {
     }
 };
 
+/**
+ * Visitor for transforms that can be applied either on timestamp or data
```

Comment:
Nit: would be nice to add a comment about the meaning of the return value. I guess it is "number of years/days/months since epoch"?